src/main/java/core/basesyntax/serviceimpl/DataConverterImpl.java

```
@@ -0,0 +1,25 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.common.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (int i = 1; i < inputReport.size(); i++) {
+            String[] lineArray = inputReport.get(i).split(",");
+            String code = lineArray[0].trim();
+            String fruit = lineArray[1].trim();
+            int quantity = Integer.parseInt(lineArray[2].trim());
+
+            FruitTransaction.Operation operation = FruitTransaction.Operation.fromCode(code);
```

Mapping operation code to enum must be validated. If `fromCode(code)` returns null or throws because the code is invalid, you should catch that and throw an informative RuntimeException that mentions the invalid operation code and the offending line number/content. The requirements state operation code must be one of b/s/p/r and invalid codes must be handled. Current mapping at line 18 does not perform such validation: .